File: Python/String/37_remove_consequtive_elements.py
"""
Idea:
appending only one unique character of a type, and recursively calling the function with remaining characters
"""


class Solution:
    def removeConsecutiveCharacter(self, S):
        if len(S) == 0:
            return ""

        firstChar = S[0]
        index = 0
        while index < len(S) and S[index] == firstChar:
            index += 1  # getting to the last index of appearance of character in 1st place of string

        return firstChar + self.removeConsecutiveCharacter(S[index:])


if __name__ == '__main__':
    T = int(input())

    for tcs in range(T):
        s = input()
        ob = Solution()
        print(ob.removeConsecutiveCharacter(s))

File: C++/String/37_remove_consequtive_elements.cpp
/*
Idea:
appending only one unique character of a type, and recursively calling the function with remaining characters
*/


#include <bits/stdc++.h> 
using namespace std;



class Solution {
public:
    string removeConsecutiveCharacter(string S)
    {
        if (S.empty())
            return "";

        char firstChar = S[0];
        int index = 1;

        while (index < S.size() && S[index] == firstChar)
            index++; // getting to the last index of appearance of character in 1st place of string


        return firstChar + removeConsecutiveCharacter(S.substr(index));
    }
};



int main()
{
    int t = 1;
    //cin >> t;
    while (t--)
    {
        string s = "aabba";
        //cin >> s;
        Solution ob;
        cout << ob.removeConsecutiveCharacter(s) << endl;
    }
}


